libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -853,17 +853,59 @@ - (NSString *)scopeQueryParamString {
  - communityId
  - communityUrl
  */
-
 - (void) updateCredentials:(NSDictionary *) params {
+    NSMutableDictionary *changeSet = [NSMutableDictionary new];
+    if (params[kSFOAuthAccessToken]) {
+        if(![params[kSFOAuthAccessToken] isEqualToString:self.credentials.accessToken])
+            changeSet[kSFOAuthAccessToken] = @[self.credentials.accessToken?
+                    self.credentials.accessToken:[NSNull null],
+                    params[kSFOAuthAccessToken]];
+
+        self.credentials.accessToken = params[kSFOAuthAccessToken];
+    }
+
+    if (params[kSFOAuthIssuedAt]) {
+        self.credentials.issuedAt = [[self class] timestampStringToDate:params[kSFOAuthIssuedAt]];
+    }
+
+    if (params[kSFOAuthInstanceUrl]) {
+        NSURL *newInstanceUrl = [NSURL URLWithString:params[kSFOAuthInstanceUrl]];
+
+        if(![newInstanceUrl isEqual:self.credentials.instanceUrl])
```

Comment:
The lack of parenthesis around these conditional blocks creeps me out, especially with the subsequent inline conditionals used.  We all remember the OpenSSL bug...